lib/index.js

```diff
@@ -10,23 +10,41 @@ const Joi = require('joi');
 const Schema = require('./schema');
 const Package = require('../package.json');
 
-// const SchwiftyMigration = require('schwifty-migration');
-
 const internals = {};
 
 exports.register = function (server, options, next) {
 
+    Joi.assert(options, Schema.plugin, 'Bad plugin options passed to schwifty.');
+
     const rootState = internals.state(server.root);
 
-    internals.state(server).dir = options.dir;
     const knexFile = options.knexFile;
 
-    internals.state(server).knex = Knex(knexFile[process.env.NODE_ENV]);
+    /*
+        Move the models into the knexFile object
+        Since a models path isn't part of it's schema on knexjs.org
+    */
+
+    if (options.models) {
+        knexFile.models = options.models;
+    }
+
+    let knex;
```

Comment:
So I've confused this a couple times, the difference between keeping server state and plugin state separated. So the way to think about it is servers interact with Schwifty via the register callback (also calling the schwifty func), and plugins interact / 'register' their input via the `schwifty` function. In this case, when we decorate the server with `server.knex`, we don't want a plugin user to be using the wrong knex instance. We can decorate `request.knex` in a plugin with the correct one, so how do we handle this?